src/hotspot/share/nmt/virtualMemoryTracker.cpp

```diff
@@ -621,6 +628,7 @@ class SnapshotThreadStackWalker : public VirtualMemoryWalker {
   SnapshotThreadStackWalker() {}
 
   bool do_allocation_site(const ReservedMemoryRegion* rgn) {
+    assert_lock_strong(NmtVirtualMemory_lock);
```

Comment:
So this is past bootstrapping?

Reply:
`MemTracker::initialize();` is called before Mutexes are ready. So it seems like that means this `assert_lock_strong` can be called before Mutexes can be used. So we do need that check for `MemTracker::is_bootstrapping_done()`.